tests/test_data_source_connectors.py

```
@@ -0,0 +1,470 @@
+def test_google_sheet_connector_is_registered():
+    # importing the module self-registers the connector under "google_sheet"
+    import app.services.google.sheets_connector  # noqa: F401
+
+    assert get_connector("google_sheet") is not None
+
+
+@pytest.mark.asyncio
+async def test_fetch_uses_key_as_tab(monkeypatch, sheet_connector):
+    captured: Dict[str, Any] = {}
+
+    async def _fake_fetch_formatted(*, spreadsheet_id, sheet_name, columns, format):
```

Ruff (0.15.17)

[error] Function argument `format` is shadowing a Python builtin

(A002)